tests/src/test/java/io/smallrye/context/test/cdi/providedBeans/BeanWithCurrentContextMethods.java

```diff
@@ -0,0 +1,83 @@
+import io.smallrye.context.SmallRyeContextManagerProvider;
+import io.smallrye.context.SmallRyeThreadContext;
+import io.smallrye.context.api.CurrentThreadContext;
+import io.smallrye.context.test.MyContext;
+
+@ApplicationScoped
+public class BeanWithCurrentContextMethods {
+
+    @CurrentThreadContext
+    public void assertCurrentContext() throws InterruptedException, ExecutionException {
+        // default is to propagate everything
+        SmallRyeThreadContext allTC = SmallRyeThreadContext.getCurrentThreadContext();
+        Assert.assertNotNull(allTC);
+        MyContext ctx = new MyContext();
+        MyContext.set(ctx);
+        Assert.assertEquals(ctx, MyContext.get());
+
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+        try {
+            // all propagated contexts
+            Runnable r = allTC.contextualRunnable(() -> {
+                Assert.assertEquals(ctx, MyContext.get());
+                Assert.assertEquals(allTC, SmallRyeThreadContext.getCurrentThreadContext());
+            });
+            executorService.submit(r).get();
+        } finally {
+            MyContext.clear();
+        }
+    }
+
+    @CurrentThreadContext(propagated = {})
+    public void assertCurrentContextAllCleared() throws InterruptedException, ExecutionException {
+        // default is to propagate nothing
+        SmallRyeThreadContext noTC = SmallRyeThreadContext.getCurrentThreadContext();
+        Assert.assertNotNull(noTC);
+        MyContext ctx = new MyContext();
+        MyContext.set(ctx);
+        Assert.assertEquals(ctx, MyContext.get());
+
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+        try {
+            // no propagated contexts
+            Runnable r = noTC.contextualRunnable(() -> {
+                Assert.assertNull(MyContext.get());
+                Assert.assertEquals(noTC, SmallRyeThreadContext.getCurrentThreadContext());
+            });
+            executorService.submit(r).get();
+        } finally {
+            MyContext.clear();
+        }
+    }
+
+    @CurrentThreadContext(remove = true)
+    public void assertCurrentContextRemoved() throws InterruptedException, ExecutionException {
+        // no current context
+        Assert.assertNull(SmallRyeThreadContext.getCurrentThreadContext());
+    }
+
+    public void assertNoCurrentContext() throws InterruptedException, ExecutionException {
+        // no current context
+        Assert.assertNull(SmallRyeThreadContext.getCurrentThreadContext());
+
+        SmallRyeThreadContext clearedTC = SmallRyeContextManagerProvider.getManager().allClearedThreadContext();
+        // make sure we can also remove it via the interceptor
+        try (CleanAutoCloseable ac = SmallRyeThreadContext.withThreadContext(clearedTC)) {
+            Assert.assertEquals(clearedTC, SmallRyeThreadContext.getCurrentThreadContext());
+            // this.calls are not intercepted in Weld 
```

Comment:
Just FYI, in normal-scoped beans you can perform self-injection and invoke methods via proxy, that should trigger interception as well.

Reply:
Do you think that would be better?

Reply:
I think it doesn't really matter, it was just an FYI since I saw the comment ;-)